### ISSUE TITLE
Bug in Panzer_ResponseScatterEvaluator_Probe_impl.hpp

### ISSUE THREAD

**Issue by bathmatt** [trilinos/Trilinos#1239]
@etphipp you wrote the initial version of this file. and @rppawlo @jmgate have modified it and I'm not clear on the logic for it but there is a definite bug in this routine.

Line 191 says
    physical_points(0,0,i) = physical_points_cell(0,i);
but 
 physical_points_cell has rank 3 dynamic rank view.
  Kokkos::DynRankView<double,PHX::Device> physical_points_cell("physical_points_cell", this->wda(d).cell_vertex_coordinates.extent_int(0), num_points, num_dim);

I'm adding an extra 0 in the access but not sure this is the right fix.  Assuming it isn't, can someone who understands it check it out?

**Comment by etphipp** [comment]
It looks like physical_points_cell was changed from rank-2 to rank-3 in https://github.com/trilinos/Trilinos/commit/74137585af04232b37db17e485e12527534594b2

As to why, @rppawlo will have to answer that.  As far as I can tell reading the code, adding the extra 0 at the beginning like you did looks correct.

**Comment by bathmatt** [comment]
It's just odd, because the array sizes are not 1, 1, dim.  so 0,0,dim seems
odd to me..  That's what I didn't understand.
We can wait til Roger looks at it though

**Comment by rppawlo** [comment]
Taking a look at this now. There were multiple implementations of check pointwise inclusion in intrepid2. When Intrepid2 was refactored to use DynRankView, we lost all but one implementation. If I recall correctly, the remaining implementation used more memory and was less efficient because it searched every element and stored results for every point in every element. We had to change our view sizes to fit the new function and may have broken something here.

**Comment by rppawlo** [comment]
Figured this out. The checkPointwiseInclusion() function requires a separate point for each cell. So to check all cells in a workset for the same point, we had to allocate physical_points_cell to be the size of number of cells and copy the same point into the array for each cell index. The array sizes for physical_point_cell are correct but are really a hack so that we can still call checkPointwiseInclusion() after the intrepid2 refactor. We could simplify this by doing the mapping to reference frame manually and then directly calling checkPointInclusion() instead of checkPointwiseInclusion(). Since this is working, not sure if changing it is the best use of my time. Any preference @bathmatt @etphipp ? Also, there are no unit tests in Panzer for this object, only in Drekar. Turns out I documented this last year in #864.
